frontend/ui/widget/inputtext.lua

```diff
@@ -422,14 +422,55 @@ function InputText:focus()
     self._frame_textwidget.color = Blitbuffer.COLOR_BLACK
 end
 
+-- Handle real keypresses from a physical keyboard, even if the virtual keyboard
+-- is shown. Mostly likely to be in the emulator, but could be Android + BT
+-- keyboard, or a "coder's keyboard" Android input method.
+function InputText:onKeyPress(key)
+
+    if key["Backspace"] then
+        self:delChar()
+    elseif key["Del"] then
+        self:rightChar()
+        self:delChar()
+    elseif key["Left"] then
+        self:leftChar()
+    elseif key["Right"] then
+        self:rightChar()
+    elseif key["End"] then
+        self:goToEnd()
+    elseif key["Home"] then
+        self:goToHome()
+    elseif key["Ctrl"] and not key["Shift"] and not key["Alt"] then
+        if key["U"] then
+            self:delToStartOfLine()
+        elseif key["H"] then
+            self:delChar()
+        end
+    else
+        return false
+    end
+
+    return true
+end
+
+-- Handle text coming directly as text from the Device layer (eg. soft keyboard
+-- or via SDL's keyboard mapping).
+function InputText:onTextInput(text)
+    self:addChars(text)
+    return true
+end
+
 function InputText:onShowKeyboard(ignore_first_hold_release)
+    Device:startTextInput()
```

Comment:
> Avoid a layer of indirection via VirtualKeyboard - there's
actually nothing specific to the VK, it could equally be used
with a PhysicalKeyboard.

Wouldn't it make more sense to duplicate it in both types of keyboard than in every single type of potential field that takes text input? I suppose all the `InputDialog`s etc. etc. goes back to InputText, but putting it in InputText seems potentially a bit "works right now." Any thoughts @poire-z @pazos @NiLuJe?

Reply:
No real/strong thought - but I don't yet have the full picture of how all this work :)
But it feels fine to have it here, in the single module that deals with input - keyboard modules do their stuff, this is just bypassing them - when they are shown.
Also, it feels self.keyboard stays and is not reinstantiated, so having it in the keyboard module might need an additional state to keep updated.

Reply:
Alright, thanks. I'll merge it soon then.